### DISCUSSION TITLE
Usando o Dapper para efetuar InnerJoin, nao consigo tipagem da variavel.

### DISCUSSION THREAD

**Discussion by etaptecnologia** [balta-io/forum#162] [Tira Dúvidas] [answered]
@andrebaltieri 
Boa tarde, como eu faco para conseguir ter a tipagem do dado em modo de compilacao? eu efetuei exatamente como na aula, porem nao consigo enxergar as propriedades de var item.

`public void GetInstituicaoEndereco()
        {

            Instituicao inst = new Instituicao();

            string sql = 
                @"SELECT * FROM [Instituicao] 
                INNER JOIN [Endereco]
                ON[Instituicao].[EnderecoId] = [Endereco].[Id]";

            var item = _connection.Query<Instituicao, Endereco, Instituicao>(
                sql,
                (instituicao, endereco) => {
                    instituicao.Endereco = endereco;

                    return instituicao;

                }, splitOn:"Id");

            //propriedades de item nao aparece.
            item.

        }`

**Comment by etaptecnologia** [comment, marked as answer]
Resolvido, usei o foreach, o comportamento de enumeradores  é diferente de listas e vetores.

Obrigado.

**Reply by andrebaltieri** [reply]
Show @etaptecnologia, se puder, coloca a solução aqui, pode ajudar outras pessoas!

**Reply by etaptecnologia** [reply]
Solução para o problema.

```
  public Instituicao GetInstituicaoEndereco(int Id)
        {
            string sql =
                @"SELECT * FROM [Instituicao] 
                INNER JOIN [Endereco]
                ON[Instituicao].[EnderecoId] = [Endereco].[Id]
                WHERE [Instituicao].[Id] = " +Id;

            var itens = _connection.Query<Instituicao, Endereco, Instituicao>(
                sql,
                (instituicao, endereco) =>
                {
                    instituicao.Endereco = endereco;

                    return instituicao;

                }, splitOn: "Id");

            return itens.ToList().FirstOrDefault();
        }
```
